packages/dds/merge-tree/src/segmentInfos.ts

```diff
@@ -159,8 +159,13 @@ export const assertMergeNode: <T extends Partial<IMergeNodeInfo> | undefined>(
 /**
  * Removes the merge node info. This is used to remove nodes from the merge-tree.
  * @param segmentLike - The segment-like object to check.
+ * @returns This function will change the type of the provided node like to never via an assertion. This
+ * ensures no further usage of the removed merge node info is allowed. if continued use is required other
+ * type coercion methods should be use to correctly re-type the variable.
```

Comment:
```suggestion
 * type coercion methods should be used to correctly re-type the variable.
```
same in below comment block